code/modules/admin/battle_royale.dm

```diff
@@ -0,0 +1,363 @@
+	if(!item_path)
+		return
+	var/turf/target = get_safe_random_station_turf()
+	var/obj/structure/closet/supplypod/battleroyale/pod = new()
+	if(islist(item_path))
+		for(var/thing in item_path)
+			new thing(pod)
+	else
+		new item_path(pod)
+	if(force_time)
+		pod.fallDuration = force_time
+	new /obj/effect/DPtarget(target, pod)
+	if(announce)
+		priority_announce("[announce] \nExpected Drop Location: [get_area(target)]\n ETA: [force_time/10] Seconds.", "High Command Supply Control")
+
+//==================================
+// WORLD BORDER
+//==================================
+
+/datum/proximity_monitor/advanced/battle_royale
+	setup_edge_turfs = TRUE
+	use_host_turf = TRUE
+	field_shape = 1
+	current_range = 5
+	var/static/image/edgeturf_south = image('icons/effects/fields.dmi', icon_state = "projectile_dampen_south")
+	var/static/image/edgeturf_north = image('icons/effects/fields.dmi', icon_state = "projectile_dampen_north")
+	var/static/image/edgeturf_west = image('icons/effects/fields.dmi', icon_state = "projectile_dampen_west")
+	var/static/image/edgeturf_east = image('icons/effects/fields.dmi', icon_state = "projectile_dampen_east")
+	var/static/image/northwest_corner = image('icons/effects/fields.dmi', icon_state = "projectile_dampen_northwest")
+	var/static/image/southwest_corner = image('icons/effects/fields.dmi', icon_state = "projectile_dampen_southwest")
+	var/static/image/northeast_corner = image('icons/effects/fields.dmi', icon_state = "projectile_dampen_northeast")
+	var/static/image/southeast_corner = image('icons/effects/fields.dmi', icon_state = "projectile_dampen_southeast")
+	var/static/image/generic_edge = image('icons/effects/fields.dmi', icon_state = "projectile_dampen_generic")
+
+/datum/proximity_monitor/advanced/battle_royale/field_edge_crossed(atom/movable/AM, obj/effect/abstract/proximity_checker/advanced/field_edge/F)
+	if(isliving(AM))
+		var/mob/living/M = AM
+		M.gib()
+		to_chat(M, "<span class='warning'>You left the zone!</span>")
+
+/datum/proximity_monitor/advanced/battle_royale/setup_edge_turf(turf/T)
+	. = ..()
+	var/image/I = get_edgeturf_overlay(get_edgeturf_direction(T))
+	var/obj/effect/abstract/proximity_checker/advanced/F = edge_turfs[T]
+	F.appearance = I.appearance
+	F.invisibility = 0
+	F.mouse_opacity = MOUSE_OPACITY_TRANSPARENT
+	F.layer = 5
+
+/datum/proximity_monitor/advanced/battle_royale/proc/get_edgeturf_overlay(direction)
+	switch(direction)
+		if(NORTH)
+			return edgeturf_north
+		if(SOUTH)
+			return edgeturf_south
+		if(EAST)
+			return edgeturf_east
+		if(WEST)
+			return edgeturf_west
+		if(NORTHEAST)
+			return northeast_corner
+		if(NORTHWEST)
+			return northwest_corner
+		if(SOUTHEAST)
+			return southeast_corner
+		if(SOUTHWEST)
+			return southwest_corner
+		else
+			return generic_edge
+
+//Checks tick when recalculating becuase field will be large
+/datum/proximity_monitor/advanced/battle_royale/update_new_turfs()
```

Comment:
Why do you actually track a list of valid turfs rather than just the coordinates?

Reply:
Whoever made fields made it track turfs instead of coordinates

Reply:
Why are you using proper fields for this?

Reply:
It's easy to have a visible field that detects when it's being passed